src/core/datastore/processors/offline-data-processor.js

```diff
@@ -63,7 +64,9 @@ export class OfflineDataProcessor extends DataProcessor {
   _processClear(collection, query, options) {
     return this._syncManager.clearSync(collection, query)
       .then(() => this._getRepository())
-      .then(repo => repo.clear(collection, query, options));
+      // TODO: from the public API, it seems that clear is just a delete with no filter
```

Comment:
Re: your comment - the difference between `clear` and a `delete` is the following:
- A `delete` also implies that the entities will be deleted from the backend (in case of a sync store, this implies we'll keep track of the `delete` in the sync queue and apply it to the backend on a subsequent `push` or `sync`).
- A `clear` is only supposed to clear local data. It has no effect on the backend. A `clear` does not impact the sync queue.

Reply:
I meant it more from an internal standpoint. Because `offlineRepo.clear()` can be more performant than an `offlineRepo.delete()`, but in order to return the count, currently I do a `delete()`.

The second part may be a misunderstanding. Currently, I clear the sync queue. Why leave the sync queue, if we are deleting the entities? We can't possibly sync the updates or creates (also, for create the sync queue is being cleared in the current implementation) upon push, so what is the use of the sync items?